### DISCUSSION TITLE
Healthcheck in pulsar-admin rest api

### DISCUSSION THREAD

**Discussion by agrahul89** [apache/pulsar#17654] [General]
Is healthcheck endpoint not available pulsar-admin rest api similar to what is available with command line pulsar-admin?
> pulsar-admin brokers healthcheck

Only rest endpoints I am able to see related to broker are - 
[/admin/broker-stats/allocator-stats/:allocator](https://pulsar.apache.org/admin/broker-stats/allocator-stats/:allocator)
[/admin/broker-stats/bookieops](https://pulsar.apache.org/admin/broker-stats/bookieops)
[/admin/broker-stats/broker-resource-availability/:property/:cluster/:namespace](https://pulsar.apache.org/admin/broker-stats/broker-resource-availability/:property/:cluster/:namespace)
[/admin/broker-stats/broker-resource-availability/:tenant/:namespace](https://pulsar.apache.org/admin/broker-stats/broker-resource-availability/:tenant/:namespace)
[/admin/broker-stats/destinations](https://pulsar.apache.org/admin/broker-stats/destinations)
[/admin/broker-stats/load-report](https://pulsar.apache.org/admin/broker-stats/load-report)
[/admin/broker-stats/mbeans](https://pulsar.apache.org/admin/broker-stats/mbeans)
[/admin/broker-stats/metrics](https://pulsar.apache.org/admin/broker-stats/metrics)
[/admin/broker-stats/topics](https://pulsar.apache.org/admin/broker-stats/topics)
[/admin/brokers/configuration](https://pulsar.apache.org/admin/brokers/configuration)
[/admin/brokers/configuration/values](https://pulsar.apache.org/admin/brokers/configuration/values)
[/admin/brokers/configuration/:configName/:configValue](https://pulsar.apache.org/admin/brokers/configuration/:configName/:configValue)
[/admin/brokers/internal-configuration](https://pulsar.apache.org/admin/brokers/internal-configuration)
[/admin/brokers/:cluster](https://pulsar.apache.org/admin/brokers/:cluster)
[/admin/brokers/:cluster/:broker/ownedNamespaces](https://pulsar.apache.org/admin/brokers/:cluster/:broker/ownedNamespaces)

**Comment by michaeljmarshall** [comment]
> Is healthcheck endpoint not available pulsar-admin rest api similar to what is available with command line pulsar-admin?

The Pulsar Admin client uses the REST API for all commands, so anything available to the admin client is available to as a rest endpoint. The path is `/admin/v2/brokers/health`. The path is included in the swagger.json file for 2.10.1, and likely others, but this is the only one I checked: https://pulsar.apache.org/swagger/2.10.1/v2/swagger.json.

**Reply by agrahul89** [reply]
Thanks for sharing the swagger spec Michael. I could not find the latest doc for the list of rest endpoints so the link you shared helps a ton.
